Development/src/pages/devices/ChannelMappingMatrix.js

```diff
@@ -0,0 +1,514 @@
+                        </SingleFieldList>
+                    </div>
+                </ReferenceManyField>
+            </div>
+            <Typography variant="body1">{'Senders:'} </Typography>
+            <div
+                style={{
+                    margin: 2,
+                    padding: 1,
+                }}
+            >
+                <ReferenceManyField
+                    record={output_item}
+                    basePath="/flows"
+                    label="Flows"
+                    source="source_id"
+                    reference="flows"
+                    target="source_id"
+                >
+                    <SingleFieldList>
+                        <ReferenceManyField
+                            label="Senders"
+                            basePath="/senders"
+                            source="id"
+                            target="flow_id"
+                            reference="senders"
+                            link="show"
+                        >
+                            <div
+                                style={{
+                                    margin: 5,
+                                    padding: 10,
+                                }}
+                            >
+                                <SingleFieldList linkType="show">
+                                    <ChipConditionalLabel source="label" />
+                                </SingleFieldList>
+                            </div>
+                        </ReferenceManyField>
+                    </SingleFieldList>
+                </ReferenceManyField>
+            </div>
+        </>
+    );
+};
+
+const OutputSourceAssociation = ({ outputs, isExpend }) =>
+    outputs.map(([output_name, output_item]) => (
+        <TableCell
+            align="center"
+            colSpan={isExpend(output_name) ? output_item.channels.length : 1}
+        >
+            {get(output_item, `source_id`) ? (
+                <Tooltip
+                    interactive
+                    title={get_source_tooltip(output_item)}
+                    placement="bottom"
+                    link
+                >
+                    <div>
+                        <ReferenceField
+                            record={output_item}
+                            basePath="/sources"
+                            label="Sources"
+                            source="source_id"
+                            reference="sources"
+                            link="show"
+                        >
+                            <ChipConditionalLabel source="label" />
+                        </ReferenceField>
+                    </div>
+                </Tooltip>
+            ) : null}
+        </TableCell>
+    ));
+
+const InputSourceAssociation = ({ isRowExpanded, input_item }) => (
+    <TableCell
+        align="center"
+        rowSpan={isRowExpanded ? input_item.channels.length + 1 : 1}
+    >
+        {input_item.parent.type === 'source' ? (
```

Comment:
In the call today, we discussed adding a tooltip here with "Type: Source" or "Type: Receiver".